### PR TITLE
Fixed some problem I discovered while working on another story

### PR DESCRIPTION
**Pull request by sollecitom**
- Fixed some problems with error handling for Observables.
- Eliminated incorrect double `stop()` call for RpcBroker.
- Added `Schedulers.shutdown()` call in `stop()` implementation for Node and Driver, to avoid stuck processes when observable pipelines go wrong.

### CODE REVIEW

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt [hunk 1]
```diff
@@ -889,7 +889,9 @@ private class NetworkVisibilityController {
                         // Nothing to do here but better being exhaustive.
                     }
                 }
-            }
+            }, { _ ->
+                // Nothing to do on errors here.
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: What errors are we getting that we can ignore?

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt [hunk 2]
```diff
@@ -955,9 +957,10 @@ fun <DI : DriverDSL, D : InternalDriverDSL, A> genericDriver(
     val shutdownHook = addShutdownHook(driverDsl::shutdown)
     try {
         driverDsl.start()
-        return dsl(coerce(driverDsl))
+        return dsl(coerce(driverDsl)).also { Schedulers.shutdown() }
```
Comment (shamsasari) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: This should in the driver dsl `shutdown` method, but even then I think this is problematic for the reasons given above.

**Comment by shamsasari** [comment]
>Added Schedulers.shutdown() call in stop() implementation for Node and Driver, to avoid stuck processes when observable pipelines go wrong.

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt (hunk 2 above)
Comment (sollecitom) [reply testing/node-driver/src/main/kotlin/net/corda/testing/node/internal/DriverDSLImpl.kt]: Removed the `Schedulers.shutdown()` everywhere, until I come up with a better way of doing it.

**Comment by sollecitom** [comment]
@shamsasari about the `Schedulers.,shutdown()`, the idea is to run `Observable.observeOn(Schedulers)`, which allows to prevent deadlocks.
When using it that way, though, it takes explicitly shutdown of `Schedulers`, which is sadly a singleton.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 3]
```diff
@@ -432,7 +432,7 @@ fun CordaRPCOps.pendingFlowsCount(): DataFeed<Int, Pair<Int, Int>> {
                     }
                 }
             }.subscribe()
-    if (completedFlowsCount == 0) {
+    if (pendingFlowsCount == 0) {
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: I just noticed this method. Is it intended to be public API? It seems more like a support function than an end-user method.

**Pull request merged**